cmd/podman/umount.go

```diff
@@ -24,18 +36,67 @@ func umountCmd(c *cli.Context) error {
 	}
 	defer runtime.Shutdown(false)
 
+	umountAll := c.Bool("all")
 	args := c.Args()
-	if len(args) == 0 {
+	if len(args) == 0 && !umountAll {
 		return errors.Errorf("container ID must be specified")
 	}
-	if len(args) > 1 {
-		return errors.Errorf("too many arguments specified")
+	if len(args) > 0 && umountAll {
+		return errors.Errorf("when using the --all switch, you may not pass any container IDs")
 	}
 
-	ctr, err := runtime.LookupContainer(args[0])
-	if err != nil {
-		return errors.Wrapf(err, "error looking up container %q", args[0])
+	umountContainerErrStr := "error unmounting container"
+	var lastError error
+	if len(args) > 0 {
+		for _, name := range args {
+			ctr, err := runtime.LookupContainer(name)
+			if err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = errors.Wrapf(err, "%s %s", umountContainerErrStr, name)
+				continue
+			}
+
+			if err = unmountContainer(ctr); err != nil {
+				if lastError != nil {
+					logrus.Error(lastError)
+				}
+				lastError = errors.Wrapf(err, "%s %s", umountContainerErrStr, name)
+				continue
+			}
+			fmt.Printf("%s\n", ctr.ID())
+		}
+	} else {
+		containers, err := runtime.GetContainers()
+		if err != nil {
+			return errors.Wrapf(err, "error reading Containers")
+		}
+		for _, ctr := range containers {
+			mountpoint, _ := ctr.Mountpoint()
```

Comment:
It's worth checking for errors here - you could get a ErrNoSuchCtr if the container was removed from under you

Reply:
OK, I've captured the error from checking the container state, if not nil I assume the ctr isn't hip and continue on to the next container in the "all containers" list.